invokeai/app/services/shared/sqlite_migrator/migrations/migration_7.py

```diff
@@ -0,0 +1,88 @@
+import sqlite3
+
+from invokeai.app.services.shared.sqlite_migrator.sqlite_migrator_common import Migration
+
+
+class Migration7Callback:
+    def __call__(self, cursor: sqlite3.Cursor) -> None:
+        self._create_models_table(cursor)
+        self._drop_old_models_tables(cursor)
+
+    def _drop_old_models_tables(self, cursor: sqlite3.Cursor) -> None:
+        """Drops the old model_records, model_metadata, model_tags and tags tables."""
+
+        tables = ["model_records", "model_metadata", "model_tags", "tags"]
+
+        for table in tables:
+            cursor.execute(f"DROP TABLE IF EXISTS {table};")
+
+    def _create_models_table(self, cursor: sqlite3.Cursor) -> None:
+        """Creates the v4.0.0 models table."""
+
+        tables = [
+            """--sql
+            CREATE TABLE IF NOT EXISTS models (
+                id TEXT NOT NULL PRIMARY KEY,
+                hash TEXT GENERATED ALWAYS as (json_extract(config, '$.hash')) VIRTUAL NOT NULL,
+                base TEXT GENERATED ALWAYS as (json_extract(config, '$.base')) VIRTUAL NOT NULL,
+                type TEXT GENERATED ALWAYS as (json_extract(config, '$.type')) VIRTUAL NOT NULL,
+                path TEXT GENERATED ALWAYS as (json_extract(config, '$.path')) VIRTUAL NOT NULL,
+                format TEXT GENERATED ALWAYS as (json_extract(config, '$.format')) VIRTUAL NOT NULL,
+                name TEXT GENERATED ALWAYS as (json_extract(config, '$.name')) VIRTUAL NOT NULL,
+                description TEXT GENERATED ALWAYS as (json_extract(config, '$.description')) VIRTUAL,
+                source TEXT GENERATED ALWAYS as (json_extract(config, '$.source')) VIRTUAL NOT NULL,
+                source_type TEXT GENERATED ALWAYS as (json_extract(config, '$.source_type')) VIRTUAL NOT NULL,
+                source_api_response TEXT GENERATED ALWAYS as (json_extract(config, '$.source_api_response')) VIRTUAL,
+                trigger_words TEXT GENERATED ALWAYS as (json_extract(config, '$.trigger_words')) VIRTUAL,
```

Comment:
Can we go ahead and add `default_settings`?

Reply:
What is the value of making these all virtual columns?

Reply:
> Can we go ahead and add default_settings?

Sure - should it be a `TEXT` column that the frontend parses, or also structured on the backend?

Reply:
> What is the value of making these all virtual columns?

I went back and forth on this and couldn't think of any good reason not to use virtual columns. 

You get indices on them, you can add and remove them with as much ease (or difficulty) as you can with "physical" columns. Seems reasonable to me. The SQL itself is so much simpler, too. If we did encounter issues, we have a pretty solid system for migrations in place now so it wouldn't be hard to change the schema.

Anything particular you are concerned with?

Reply:
It is probably just because I am unfamiliar - but having to spread new values across the existing config in order to update a column feels not great to me when there is no real reason to jam all of this into a config object

Reply:
Ok, that's a pretty darn good reason to not use generated columns. But I think we can surgically update JSON without needing to provide the whole object. I'll investigate.

Reply:
Yeah, we can set specific JSON keys in SQLite: https://www.sqlite.org/json1.html#jset

Not complicated either, especially for this flat config object.

I noticed the update routes expect the full object, while I'd prefer a structured `ModelRecordChanges` object represented the allowed changes. Let's address this stuff in a followup.

Reply:
I figured out a simple way to do this that also validates the values. Only the `ModelRecordChanges` is passed to the `update` method.